llap-client/src/java/org/apache/hadoop/hive/registry/impl/ZkRegistryBase.java

```diff
@@ -353,6 +356,23 @@ protected final void updateServiceRecord(
     }
   }
 
+  protected void ensurePersistentNodePath(ServiceRecord srv) throws IOException {
+    try {
+      String pNodePath = "/" + PATH_JOINER.join(namespace, StringUtils.substringBetween(workersPath, "/", "/"),
+          "pnode0");
+      LOG.info("Check if persistent node  path{}  exists, create if not", pNodePath);
```

Comment:
whitespace is strange here, this should be:
```
Check if persistent node path {} exists, create if not
```

Reply:
fixed